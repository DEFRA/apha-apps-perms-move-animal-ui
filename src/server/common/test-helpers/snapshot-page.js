import { parseDocument } from '../../common/test-helpers/dom.js'
import { createServer } from '~/src/server/index.js'
import { withCsrfProtection } from './csrf.js'
/** @import { Server } from '@hapi/hapi' */

/* global expect, it, beforeAll, afterAll, describe */

/**
 * @param {string} describeMessage
 * @param {string} shouldMessage
 * @param {string} urlPath
 */
export const describePageSnapshot = (
  describeMessage,
  shouldMessage,
  urlPath
) => {
  describe(describeMessage, () => {
    /** @type {Server} */
    let server

    beforeAll(async () => {
      server = await createServer()
      await server.initialize()
    })

    afterAll(async () => {
      await server.stop({ timeout: 0 })
    })

    it(shouldMessage, async () => {
      const { payload } = await server.inject(
        withCsrfProtection({
          method: 'GET',
          url: urlPath
        })
      )

      const content =
        parseDocument(payload).querySelector('#main-content')?.innerHTML

      expect(content).toMatchSnapshot()
    })
  })
}
