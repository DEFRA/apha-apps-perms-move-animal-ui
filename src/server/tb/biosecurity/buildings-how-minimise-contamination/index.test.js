import { describePageSnapshot } from '../../../common/test-helpers/snapshot-page.js'
import { HousingOtherPage } from '../housing-other/index.js'
import { peopleDisinfectionPage } from '../people-disinfection/index.js'
import {
  Answer,
  buildingsHowMinimiseContaminationPage,
  BuildingsHowMinimiseContaminationPage
} from './index.js'

const sectionKey = 'biosecurity'
const question =
  'Which measures are being taken to reduce the spread of TB during housing?'
const questionKey = 'buildingsHowMinimiseContamination'
const pageUrl = '/biosecurity/buildings-how-minimise-contamination'

describe('BuildingsHowMinimiseContaminationPage', () => {
  const page = new BuildingsHowMinimiseContaminationPage()

  it('should have the correct urlPath', () => {
    expect(page.urlPath).toBe(pageUrl)
  })

  it('should have the correct sectionKey', () => {
    expect(page.sectionKey).toBe(sectionKey)
  })

  it('should have the correct question', () => {
    expect(page.question).toBe(question)
  })

  it('should have the correct questionKey', () => {
    expect(page.questionKey).toBe(questionKey)
  })

  it('should have the correct title', () => {
    expect(page.title).toBe(question)
  })

  it('should have the correct Answer model', () => {
    expect(page.Answer).toBe(Answer)
  })

  it('nextPage should return people disinfection page', () => {
    const answer = new page.Answer({ [questionKey]: ['isolation'] })
    const nextPage = page.nextPage(answer)
    expect(nextPage).toBe(peopleDisinfectionPage)
  })

  it('nextPage should return housingOtherPage', () => {
    const answer = new page.Answer({ [questionKey]: ['other'] })
    const nextPage = page.nextPage(answer)
    expect(nextPage).toBeInstanceOf(HousingOtherPage)
  })

  it('should export page', () => {
    expect(buildingsHowMinimiseContaminationPage).toBeInstanceOf(
      BuildingsHowMinimiseContaminationPage
    )
  })

  describePageSnapshot({
    describes: 'buildingsHowMinimiseContaminationPage.content',
    it: 'should render expected response and content',
    pageUrl
  })
})
