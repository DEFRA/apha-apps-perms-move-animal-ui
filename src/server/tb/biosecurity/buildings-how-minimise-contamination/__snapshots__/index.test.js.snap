// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingsHowMinimiseContaminationPage buildingsHowMinimiseContaminationPage.content should render expected response and content 1`] = `
"
    <form method="POST" novalidate="">
      <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
          <input type="hidden" name="crumb" value="csrf-value">
          <input type="hidden" name="nextPage" value="">

<div class="govuk-form-group">
  <fieldset class="govuk-fieldset" aria-describedby="buildingsHowMinimiseContamination-hint">
    <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
      <h1 class="govuk-fieldset__heading">
        Which measures are being taken to reduce the spread of TB during housing?
      </h1>
    </legend>
    <div id="buildingsHowMinimiseContamination-hint" class="govuk-hint">
      Only select the measures you are taking. You do not need to select them all to receive your licence.
    </div>
    <div class="govuk-checkboxes" data-module="govuk-checkboxes">
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input" id="buildingsHowMinimiseContamination" name="buildingsHowMinimiseContamination" type="checkbox" value="cleaning" data-testid="cleaning-checkbox">
        <label class="govuk-label govuk-checkboxes__label" for="buildingsHowMinimiseContamination">
          Cleaning and disinfecting of shared buildings before the animals arrive
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input" id="buildingsHowMinimiseContamination-2" name="buildingsHowMinimiseContamination" type="checkbox" value="isolation" data-testid="isolation-checkbox">
        <label class="govuk-label govuk-checkboxes__label" for="buildingsHowMinimiseContamination-2">
          Isolation of incoming animals in a separate building, shed or pen on arrival
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input" id="buildingsHowMinimiseContamination-3" name="buildingsHowMinimiseContamination" type="checkbox" value="incoming" data-testid="incoming-checkbox">
        <label class="govuk-label govuk-checkboxes__label" for="buildingsHowMinimiseContamination-3">
          Incoming animals will be permanently housed
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input" id="buildingsHowMinimiseContamination-4" name="buildingsHowMinimiseContamination" type="checkbox" value="disinfection" data-testid="disinfection-checkbox">
        <label class="govuk-label govuk-checkboxes__label" for="buildingsHowMinimiseContamination-4">
          Disinfection points located at the housing entrances and exits
        </label>
      </div>
      <div class="govuk-checkboxes__item">
        <input class="govuk-checkboxes__input" id="buildingsHowMinimiseContamination-5" name="buildingsHowMinimiseContamination" type="checkbox" value="other" data-testid="other-checkbox">
        <label class="govuk-label govuk-checkboxes__label" for="buildingsHowMinimiseContamination-5">
          Other measures
        </label>
      </div>
    </div>
  </fieldset>
</div>

            <button type="submit" class="govuk-button" data-module="govuk-button" id="continue-button">
  Continue
</button>
        </div>
      </div>
    </form>
        "
`;
